### PR TITLE
Allow chpldoc to respond to private symbols

### PR DESCRIPTION
**Pull request by lydia-duncan**
I just searched for all places where the pragma "no doc" meant a symbol
was ignored, and replaced them with a call to a method on the symbol class
which checks if it has private or "no doc" attached to it.

Includes a test that select symbols that can be private are ignored by chpldoc,
and two tests on future private support.  Since the support I added for chpldoc
encompassed all places where "no doc" could be applied, it is expected that
when type aliases, enums, classes/records, fields, and method can support the
private keyword, these tests will start working.

### CODE REVIEW

**Comment by lydia-duncan** [comment]
Passed chpldoc testing (test/chpldoc/, test/compflags/thomasvandoren/, test/release/examples/primers/chpldoc.doc.chpl)

**Comment by bradcray** [comment]
I took only the briefest of glances at this before realizing that it was bigger than I have time to look at just now, but my quick reactions are:

* could the condition to squash documentation be put into a method or helper function so that if/when additional conditions come on-line, one spot could be edited rather than a dozen or two?

* is it still the case that our chpldoc testing is testing the text mode that has nothing to do with the HTML output?  If so, have you done any spot-checking (or diffing) of the generated HTML to see that it matches the output prior to this change?

**Comment by bradcray** [comment]
(Also, could/should this commit remove no-longer-necessary "no doc" pragmas from modules/standard?)

**Comment by lydia-duncan** [comment]
"could the condition to squash documentation be put into a method or helper function so that if/when additional conditions come on-line, one spot could be edited rather than a dozen or two?"
Yup!  I should have thought of that

"is it still the case that our chpldoc testing is testing the text mode that has nothing to do with the HTML output? If so, have you done any spot-checking (or diffing) of the generated HTML to see that it matches the output prior to this change?"
Yup and yup!  I'm doing that as I remove the "no doc" pragmas from modules/standard.

Since the addition of private is still in progress (i.e., still being tested, but all committed to branches for review by the documentation authors), I figured I wouldn't hold up this feature going in by tying it to the removal of the pragmas.  I have a separate branch for taking care of that (which I don't intend to have reviewed, because it seems like not a good use of a reviewer's time, given how obvious of a change it is).

File: test/chpldoc/nodoc/privateClasses.doc.catfiles [hunk 1]
```diff
@@ -0,0 +1 @@
+docs/privateClasses.txt
```
Comment (vasslitvinov) [inline test/chpldoc/nodoc/privateClasses.doc.catfiles]: Please add a newline at the end. (Unless there is a strong reason not to.)

File: test/chpldoc/nodoc/privateTypeAlias.doc.catfiles [hunk 2]
```diff
@@ -0,0 +1 @@
+docs/privateTypeAlias.txt
```
Comment (vasslitvinov) [inline test/chpldoc/nodoc/privateTypeAlias.doc.catfiles]: Ditto for newline.

**Comment by vasslitvinov** [comment]
OK to merge, after adding newlines.

**Pull request merged**